### ISSUE TITLE
Multi block selection: Block Switcher misleading label and tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51917]
### Description

When selecting multiple block, the Block toolbar shows commands that are applicable to the set of selected blocks. So far so good.

In the toolbar, the Block Switched button shows the ‘copy’ icon. Could be better, but yes: so far so good.
However, the aria-label and the tooltip text refer to the first block title. Not sure it makes any sense. I would expect it would tell users it’s a multi-selection state or something along those lines.

The label and tooltip actually are the title of the first block in the set of selected blocks:

<img width="1500" alt="Screenshot 2023-06-26 at 11 44 34" src="https://github.com/WordPress/gutenberg/assets/1682452/0d874a85-5c03-4406-a4ee-4207b1b3ce89">

Also, the Block switcher button has a visually hidden description, associated by the means of `aria-describedby`. The text of the description does refer to the multi-selection state and it's `Change type of %d blocks` but it doesn't make sense in combination with the first block title.

Actually, screen reader users will hear something like:

`Quote. Change type of 3 blocks`

Turns out with multi-selection the label of the Block switcher button used to be `Change type of %d blocks`. It was changed to the first block title in https://github.com/WordPress/gutenberg/pull/26135

The current labeling + description doesn't make sense. I would expect it to be something like:

Label: Multiple blocks selected
Description: Change type of %d blocks

It should be more concise that that, because the label becomes visible when the preference 'Show button text labels' is enabled:

<img width="725" alt="Show button text labels" src="https://github.com/WordPress/gutenberg/assets/1682452/252a3757-2f7a-4ec1-89a8-c19dbdfad8df">

[Looking at the code](https://github.com/WordPress/gutenberg/blob/48b30f4b7da156293b4ffd568db4f879eacde852/packages/block-editor/src/components/block-switcher/index.js#L165-L182), the description was made conditional and changes depending on whether there's a multi-block selection. Instead, the label is always the block title, which doesn't seem correct. It should change conditionally as well.

## Additionally:
I'd also suggest to improve a bit the labeling + description for the single selection state. In this case, the Description repeats the block title, which is redundant and just adds noise:

<img width="566" alt="Screenshot 2023-06-26 at 12 07 26" src="https://github.com/WordPress/gutenberg/assets/1682452/6c8bf05b-d503-42c2-b0f9-a30873c32745">

### Step-by-step reproduction instructions

- Create a new post.
- Add a few blocks of different type.
- Click in a Paragraph block.
- Select all the blocks by pressing Cmd/Ctrl + A twice.
- The block toolbar appears.
- Hover the Block switcher button in the toolbar (the first button).
- Observe the tooltip text is the title of the first block.
- Unselect the blocks.
- Change order of the blocks so that the first one is of a different type.
- Select all again.
- Hover the Block switcher button in the toolbar.
- Observe the tooltip text is changed because it's the title of the new first block.
- Inspect the source or use a screen reader to check the visually hidden description associated with the Block switcher button.
- Observe the description text is `Change type of %d blocks`.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
PR #54692.
